modules/cli/apply/apply.go

```diff
@@ -90,18 +94,50 @@ func Apply(ctx context.Context, client *client.Getter, name string, baseDirs []s
 						return nil
 					}
 				}
-				if err := Dir(ctx, client, name, path, opts); err == ErrNoResources {
+				if err := Dir(ctx, client, name, path, opts, gitRepoBundlesMap); err == ErrNoResources {
 					logrus.Warnf("%s: %v", path, err)
 					return nil
 				} else if err != nil {
 					return err
 				}
 				foundBundle = true
+
 				return nil
 			})
 			if err != nil {
 				return err
 			}
+
+			logrus.Infof("gitRepoBundlesMap %v", gitRepoBundlesMap)
+			//list all bundles for this gitrepo and prune those not found
+			c, err := client.Get()
+			if err != nil {
+				return err
+			}
+
+			gitRepo, err := c.Fleet.GitRepo().Cache().Get(client.Namespace, name)
+			if err != nil {
+				return err
+			}
+			logrus.Infof("gitRepo %v", gitRepo.Spec)
+			if gitRepo.Spec.PruneBundles {
+				filter := labels.Set(map[string]string{fleet.RepoLabel: name})
+				bundles, err := c.Fleet.Bundle().List(client.Namespace, metav1.ListOptions{LabelSelector: filter.AsSelector().String()})
+				logrus.Infof("Bundles for repo %v", bundles.Items)
+				if err != nil {
+					return err
+				}
+
+				for _, bundle := range bundles.Items {
+					if ok := gitRepoBundlesMap[bundle.Name]; !ok {
+						logrus.Infof("Bundle to be deleted %v %v", bundle.Namespace, bundle.Name)
+						err := c.Fleet.Bundle().Delete(bundle.Namespace, bundle.Name, nil)
+						if err != nil {
+							return err
+						}
+					}
+				}
+			}
```

Comment:
we could consider refactoring this whole logic into using wrangler.apply, but as spoken offline I'm not sure how much work that would be

Reply:
yeah, that would help by not having two separate create and delete calls via API; however yeah maybe this can be added as an improvement later